python_training/sim_city.py

```diff
@@ -0,0 +1,198 @@
+BASE_MUNICIPALITY_TAX_RATE = 1000
+
+
+class Ackland:
+    """
+    The state Ackland itself.
+    """
+
+    def __init__(self, cities):
+        """
+        Initiates the Ackland state object.
+        """
+        if type(cities) is City:
+            cities = [cities]
+        self.cities = cities
+
+    def calculate_total_tax_amount(self) -> int:
+        """
+        Calculates the amount of tax money that would be earned from all the cities of Ackland.
+
+        :return: The total amount of tax money from all the cities of the state.
+        """
+        return sum([city.how_much_money() for city in self.cities])
+
+
+class City:
+    """
+    A city in Ackland.
+    """
+
+    def __init__(self, city_name):
+        """
+        Initiates a new City object.
+
+        :param city_name: The name of the city you would like to create.
+        """
+        self.city_name = city_name
+        self.base_municipality_tax_rate = BASE_MUNICIPALITY_TAX_RATE
+        self.neighborhoods = {}
+
+    def how_much_money(self) -> int:
+        """
+        Calculates the concluded amount of tax money that would be payed by The cities in Ackland.
+
+        :return: The amount of money that should be payed.
+        """
+        return self.base_municipality_tax_rate + sum([neighborhood.get_neighborhood_tax_rate() for neighborhood
+                                                      in self.neighborhoods.values()])
+
+    def ruin_a_neighborhood(self, neighborhood_name):
+        """
+        Removes a neighborhood from the city.
+
+        :param neighborhood_name: The name of the neighborhood that would be removed.
+        """
+        if neighborhood_name in self.neighborhoods:
+            self.base_municipality_tax_rate *= 1.05
+            self.neighborhoods.pop(neighborhood_name)
+        else:
+            raise KeyError(f"The neighborhood {neighborhood_name} doesnt exist in {self.city_name}")
+
+    def build_a_neighborhood(self, neighborhood_name):
+        """
+        Adds a neighborhood to the city.
+
+        :param neighborhood_name: The name of the neighborhood.
+        :return A Neighborhood object if succeeded, None otherwise.
+        """
+        if neighborhood_name not in self.neighborhoods:
+            self.base_municipality_tax_rate *= 1.1
+            self.neighborhoods[neighborhood_name] = Neighborhood(neighborhood_name)
+            return self.neighborhoods[neighborhood_name]
+        else:
+            print(f"The neighborhood {neighborhood_name} already exist in {self.city_name}")
+
+    def build_a_house(self, neighborhood_name, number_of_family_members, size_of_house):
+        """
+        Adds a house to a neighborhood.
+
+        :param neighborhood_name: The neighborhood that a house would be added to.
+        :param number_of_family_members: The number of family members in the house.
+        :param size_of_house: The size of the house.
+        """
+        if neighborhood_name in self.neighborhoods:
```

Comment:
Maybe extract neighbourhood verification to a small private method, since it repeats a lot?
This can be nice for future changes, for example if you'll decide to ignore case and then use `lower()` on the given neighbourhood name every time.

Reply:
Good idea! :)